src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get, set } from 'lodash';
+
+export default {
+  visualization: {
+    '7.0.0': (doc) => {
+      const savedSearchId = get(doc, 'attributes.savedSearchId');
+      // Set new "references" attribute
+      doc.references = doc.references || [];
+      // Migrate index pattern
```

Comment:
Mind putting this shared section of code into a helper function at the top of this file?